testing/src/test/java/io/stargate/it/cql/DataTypeTest.java

```diff
@@ -0,0 +1,279 @@
+import com.datastax.oss.driver.api.core.type.TupleType;
+import com.datastax.oss.driver.api.core.type.UserDefinedType;
+import com.datastax.oss.driver.api.core.type.reflect.GenericType;
+import com.datastax.oss.driver.api.querybuilder.schema.CreateTable;
+import com.datastax.oss.driver.internal.core.type.UserDefinedTypeBuilder;
+import com.datastax.oss.protocol.internal.util.Bytes;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import io.stargate.it.storage.ClusterConnectionInfo;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.nio.ByteBuffer;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.StringJoiner;
+import java.util.UUID;
+import java.util.concurrent.atomic.AtomicInteger;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.Test;
+import org.opentest4j.AssertionFailedError;
+
+public class DataTypeTest extends JavaDriverTestBase {
+
+  public DataTypeTest(ClusterConnectionInfo backend) {
+    super(backend);
+  }
+
+  @Test
+  public void should_write_and_read_all_types() {
+
+    List<TypeSample<?>> types = generateAllTypes(keyspaceId);
+
+    createSchema(types);
+
+    // This should be a Junit parameterized test, but because of the way BaseOsgiIntegrationTest is
+    // implemented, it would start a new Stargate container for each execution.
+    // TODO refactor BaseOsgiIntegrationTest into a Junit extension
+    for (TypeSample<?> sample : types) {
+      try {
+        should_write_and_read(sample);
+      } catch (Throwable t) {
+        throw new AssertionFailedError(
+            String.format(
+                "Error on %s (cqlType=%s, value=%s)",
+                sample.columnName,
+                sample.cqlType.asCql(true, true),
+                session
+                    .getContext()
+                    .getCodecRegistry()
+                    .codecFor(sample.cqlType)
+                    .format(sample.value)),
+            t);
+      }
+    }
+  }
+
+  private void createSchema(List<TypeSample<?>> types) {
+    // Creating a new table for each type is too slow, use a single table with all possible types:
+    CreateTable createTableQuery = createTable("test").withPartitionKey("k", DataTypes.INT);
+    for (TypeSample<?> sample : types) {
+      createTableQuery = createTableQuery.withColumn(sample.columnName, sample.cqlType);
+      if (sample.cqlType instanceof UserDefinedType) {
+        session.execute(((UserDefinedType) sample.cqlType).describe(false));
+      }
+    }
+    session.execute(createTableQuery.asCql());
+  }
+
+  private <JavaTypeT> void should_write_and_read(TypeSample<JavaTypeT> sample) {
```

Comment:
Codacy found an issue: [The instance method name 'should_write_and_read' doesn't match '[a-z][a-zA-Z0-9]*'](https://app.codacy.com/gh/stargate/stargate/pullRequest?prid=6263918)